### PR TITLE
Feature :: Action Menu on DataViewer's header cells

### LINKED ISSUE #32: popover component
We need a generic popover component that will be used both on category buttons and column headers.

### PR DESCRIPTION
**Pull request by Dalanir**
That PR create the Action Menu component that position itself according to which header we have clicked on. For the moment, only "Rename Column" is active and permit to open its form in order to create this step.

I open an issue for missing tests: #157 and that PR closes #32

### CODE REVIEW

File: tests/unit/form-rename-step.spec.ts [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
-import Vuex from 'vuex';
+import Vuex, { Store } from 'vuex';
 import { setupStore } from '@/store';
+import { VQBState } from '@/store/state';
 
 const localVue = createLocalVue();
 localVue.use(Vuex);
 
-const emptyStore = setupStore({});
+// const emptyStore = setupStore({});
```
Comment (vdestraitt) [inline tests/unit/form-rename-step.spec.ts]: Can we delete this ?

**Comment by vdestraitt** [comment]
@Dalanir another thing that needs a fix is that now when selecting "rename column" in the action menu from the header of a column, `step.oldname` in the step form is not populated with the column name anymore

File: tests/unit/action-menu.spec.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { mount } from '@vue/test-utils';
+import ActionMenu from '@/components/ActionMenu.vue';
+
+describe('Action Menu', () => {
+  it('should instantiate with its popover hidden', () => {
+    const wrapper = mount(ActionMenu);
```
[the remark this answers is not in the document]
Comment (Dalanir) [reply tests/unit/action-menu.spec.ts]: I change the assertion so I have to keep it now

File: tests/unit/action-menu.spec.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    expect(wrapper.classes()).toContain('popover--active');
+  });
+
+  describe('when click on "Rename column"', () => {
+    it('should have an "Rename column" action', () => {
+      const wrapper = mount(ActionMenu);
+      const actionsWrapper = wrapper.findAll('.action-menu__option');
+
+      expect(actionsWrapper.at(1).text()).toEqual('Rename column');
```
[the remark this answers is not in the document]
Comment (Dalanir) [reply tests/unit/action-menu.spec.ts]: 👍

File: src/components/DataViewer.vue [hunk 4]
```diff
@@ -5,13 +5,24 @@
...
             :key="column.name"
-            @click="toggleColumnSelection(column.name)"
+            @click="toggleColumnSelection({ column: column.name})"
           >
             <span class="data-viewer__header-label">{{ column.name }}</span>
-            <i class="data-viewer__header-action fas fa-angle-down" @click="openStep(column.name)"></i>
+            <i
+              class="data-viewer__header-action fas fa-angle-down"
+              :class="{'data-viewer__header-action--visible': column.isActionMenuOpened}"
+              @click.stop="openMenu(index)"
```
[the remark this answers is not in the document]
Comment (Dalanir) [reply src/components/DataViewer.vue]: Yes it appears that we need it otherwise it triggers another click inside the ActionMenu :/

**Pull request merged**